src/super_gradients/training/models/segmentation_models/unet/unet_decoder.py

```diff
@@ -34,8 +35,16 @@ class UpFactorBlock(AbstractUpFuseBlock):
     Ignore Skip features, simply apply upsampling and ConvBNRelu layers.
     """
 
-    def __init__(self, in_channels: int, skip_channels: int, out_channels: int, up_factor: int, mode: str, num_repeats: int, **kwargs):
+    def __init__(self, in_channels: int, skip_channels: int, out_channels: int, up_factor: int, mode: str, num_repeats: int, fallback_mode: str, **kwargs):
         super().__init__(in_channels=in_channels, skip_channels=0, out_channels=out_channels)
+        if mode in [UpsampleMode.PIXEL_SHUFFLE.value, UpsampleMode.NN_PIXEL_SHUFFLE.value]:
+            # Check if in_channels is divisible by (up_factor ** 2) for pixel shuffle, else fallback to fallback_mode.
```

Comment:
Usually if in_channels is not divisible by (up_factor ** 2) one can do 1x1 conv from in_channels  to make it divisible and then do pixel shuffle. Do we want to support this?

Reply:
It isn't that simple, should we project to the closest floor or ceil? imagine for `input channels=2` and `scale_factor=4` does it make sense to project from 2 to 16 channels, to end with 1 channel? 
IMO the user should resolve those issues, and we shouldn't imply this. This is why I suggest a fallback mode that the user is aware of.